test/state/state.cpp

```diff
@@ -48,8 +63,19 @@ int64_t compute_tx_intrinsic_cost(evmc_revision rev, const Transaction& tx) noex
     const auto initcode_cost =
         is_create && rev >= EVMC_SHANGHAI ? initcode_word_cost * num_words(tx.data.size()) : 0;
     const auto tx_cost = is_create && rev >= EVMC_HOMESTEAD ? create_tx_cost : call_tx_cost;
-    return tx_cost + compute_tx_data_cost(rev, tx.data) + compute_access_list_cost(tx.access_list) +
-           initcode_cost;
+    auto c = tx_cost + compute_tx_data_cost(rev, tx.data) +
+             compute_access_list_cost(tx.access_list) + initcode_cost;
+
+    if (tx.type == Transaction::Type::blob)
+    {
+        static constexpr auto DATA_GAS_PER_BLOB = 0x20000;
+        const auto data_gas_price = fake_exponential(1, excess_data_gas, 3338477);
+        const auto total_data_gas = DATA_GAS_PER_BLOB * tx.blob_hashes.size();
+        const auto data_fee = total_data_gas * data_gas_price;
+        c += static_cast<int64_t>(data_fee);
```

Comment:
This is seems to be horribly wrong because we are adding a "fee" to the "gas". Maybe `data_gas_price` is always 1 in tests?